File: Python/UPCMatricula/controllers/MatriculaController.py
from models.Alumno import Alumno
from models.Curso import Curso

class MatriculaController(object):
    def __init__(self):
        self.curso = self.crear_curso("POO")

    def crear_curso(self, nombre):
        return Curso(nombre)

    def crear_alumno(self, codigo, nombre, apellido):
        return Alumno(codigo, nombre, apellido)

    def matricular(self, codigo, nombre, apellido):
        alumno = self.crear_alumno(codigo, nombre, apellido)
        self.curso.agregar_alumnos(alumno)
        return "Alumno matriculado"

    def reporte_matricula(self):
        return self.curso.listar_matriculados()

File: PHP/UPCMatricula/controllers/MatriculaController.php
<?php

class MatriculaController {

    private $curso;

    public function __construct() {
        $this->crearCurso();
    }

    public function crearCurso() {
        $this->curso = new Curso("POO");
    }

    public function crearAlumno($codigo, $nombre, $apellido) {
        $alumno = new Alumno($codigo, $nombre, $apellido);
        return $alumno;
    }

    public function matricular($codigo, $nombre, $apellido) {
        $alumno = new Alumno($codigo, $nombre, $apellido);
        $this->curso->agregarAlumno($alumno);
        return "Alumno matriculado";
    }

    public function reporteMatricula() {
        return $this->curso->listarMatriculados();
    }
}
